internal/mining/mining.go

```diff
@@ -1893,6 +1893,14 @@ mempoolLoop:
 		}
 	}
 
+	// Ensure that mining the block would not cause the chain to become
+	// unrecoverable due to ticket exhaustion.
+	err = g.chain.CheckTicketExhaustion(&best.Hash, uint8(freshStake))
+	if err != nil {
+		log.Debug(err)
```

Comment:
The idea here was that currently, no errors are logged in template creation, because there is nothing the user can do about it anyway, since the expectation is that the will only ever create valid templates.

However, this is a case (probably the only case) where the user can actually do something to resolve failure to create the template, so I thought it should be logged.

Reply:
Got it.